intervention_graph_creation/src/pipeline/local_graph.py

```diff
@@ -0,0 +1,81 @@
+from typing import List, Optional
+import numpy as np
+from pydantic import BaseModel, ConfigDict
+
+from ..local_graph_extraction.core import Node, Edge
+
+
+class GraphNode(Node):
+    """Extended Node class with embedding support."""
+    embedding: Optional[np.ndarray] = None
```

Comment:
@jefedigital  Note that we'll attach a `Metadata` subclass here to store information such as: arxiV ID, paper DOI, author etc. To my understanding, the reason you didn't define it in your schema was to avoid wasting o3's reasoning on things which can be fetched deterministically through API. 

That said, in our real database we would require to push that data in our Node. Hence, a new data structure here.

We could introduce another `Flow` to completely own defining data-structures, currently I've mixed them both in interest of simplicity.

cc: @MartinLeitgab